### PR TITLE
overlay: always mount, even if we have no parent layer

### PR DESCRIPTION
**Pull request by nalind**
Always mount an overlay filesystem, even when there's no parent layer, so that we can set an SELinux context for the mountpoints.  We create an "empty" directory to use as the lower directory for the overlay mount to satisfy the kernel, but as it's not a real layer, we don't track it in the lists of lower directories.

### CODE REVIEW

**Comment by nalind** [comment]
@rhatdan @TomSweeneyRedHat PTAL

File: drivers/overlay/overlay.go [hunk 1]
```diff
@@ -181,7 +181,7 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		return nil, fmt.Errorf("Storage option overlay.size only supported for backingFS XFS. Found %v", backingFs)
 	}
 
-	logrus.Debugf("backingFs=%s, projectQuotaSupported=%v", backingFs, projectQuotaSupported)
+	logrus.Debugf("backingFs=%s, projectQuotaSupported=%v, useNativeDiff=%v", backingFs, projectQuotaSupported, !useNaiveDiff(home))
```
Comment (TomSweeneyRedHat) [inline drivers/overlay/overlay.go]: I know not part of this change, but curious why the variable useNaiveDiff and useNaiveLock are Naive and not Native?

Comment (nalind) [reply drivers/overlay/overlay.go]: `NaiveDiffDriver` is the name of the wrapper object that several of the storage drivers use to implement their `Changes()`/`Diff()`/`DiffSize()`/`ApplyDiff()` methods, so I expect that variables related to deciding whether or not its use is necessary are named after it.  The "native" implementation in the overlay driver uses knowledge of how the overlay filesystem works to gain speed advantages, but it can't cope with certain kernel features, so that decision has to be made at startup-time.

Comment (TomSweeneyRedHat) [reply drivers/overlay/overlay.go]: Thanks nalind!

**Comment by TomSweeneyRedHat** [comment]
LGTM, would like to get another pair of eyeballs on it.

**Comment by rhatdan** [comment]
LGTM and a little simpler.
